packages/@react-aria/slider/src/useSliderThumb.ts

```diff
@@ -83,7 +91,9 @@ export function useSliderThumb(
   }, [isFocused, focusInput]);
 
   const stateRef = useRef<SliderState>(null);
-  stateRef.current = state;
+  useLayoutEffect(() => {
+    stateRef.current = state;
+  });
```

Comment:
We can get rid of `stateRef` completely if we make `useMove` use `useEffectEvent`. Done in #4564.